File: task-1/src/main/java/ru/biluta/version5/SimulationClock.java
package ru.biluta.version5;

import lombok.Getter;

@Getter
public class SimulationClock {
    private double currentTime;

    public SimulationClock() {
        this.currentTime = 0.0;
    }

    public void advanceTime(double newTime) {
        if (newTime >= currentTime) {
            currentTime = newTime;
        } else {
            throw new IllegalArgumentException("Simulation time cannot move backwards.");
        }
    }


}

File: task2/src/main/kotlin/ru.biluta.task2/SimulationClock.kt
package main.kotlin.ru.biluta.task2

data class SimulationClock(
    var currentTime: Double = 0.0
) {

    fun advanceTime(newTime: Double) {
        when (newTime >= currentTime) {
            true -> currentTime = newTime
            else -> throw IllegalArgumentException("Simulation time cannot move backwards.")
        }
    }

}